### PR TITLE
Single Host + TLS on OpenShift fixup

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
:man_facepalming: it brings changes which I made but did not push for https://github.com/devfile/devworkspace-operator/pull/201 before merging

### CODE REVIEW

File: internal/map/map.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package maputils
+
+func Append(target map[string]string, key, value string) map[string]string {
```
Comment (amisevsk) [inline internal/map/map.go:15]: One minor detail in this implementation: this approach has a side-effect if you pass in a non-nil map, which could be a little confusing:
```go
	var m map[string]string
	Append(m, "one", "two")
	fmt.Printf("%+v\n", m) // Prints nil map
```
```go
	m := map[string]string{"one": "two"}
	Append(m, "one", "three")
	fmt.Printf("%+v\n", m) // Prints map[one:three]
```
compare this to the built-in append, which does not modify the original argument.

**Review by amisevsk — COMMENTED** [review]
One minor comment

File: internal/map/map.go (hunk 1 above)
Comment (sleshchenko) [reply internal/map/map.go:15]: You're right.

I misunderstood the documentation:

> // If it has sufficient capacity, the destination is resliced to accommodate the
> // new elements.

I also considered naming like `Put(...)` or `SafelyPut(...)`
@amisevsk Any ideas on how we can avoid confusion? Do you think it's better to use any of the names above?

Comment (sleshchenko) [reply internal/map/map.go:15]: P.S. I wish golang allows enable compiler check they have for build-in append, where return value must be used or explicitly ignored.
Screenshot_20201204_110759

**Pull request merged**